packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -325,69 +325,78 @@ const numberWithUnitCellView = NumberWithUnitCellView.compose({
-            throw new Error('Type for cellData is incorrect!');
-        }
+export class ColumnInternals<TColumnConfig> {
+    ...
+    public readonly delegatedEvents: readonly string[];
+    ...
+    public constructor(options: ColumnInternalsOptions) {
+        this.delegatedEvents = options.delegatedEvents;
+        ...
     }
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
+AnchorTableColumn extends TableColumn {
+    constructor() {
+        super({
+            delegatedEvents: ['click'],
+            ...
+        });
     }
+    ...
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
 ```
 
-Angular template:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnDelegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column.
 
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
+```TS
+this.addEventListener(delegatedEventName, event => {
+    this.column.dispatchEvent(new CustomEvent('delegated-event', {
+        details: {
+            originalEvent: event
+        }
+    }));
+});
 ```
 
-Angular code:
+The `CustomEvent`'s details will be of type `DelegatedEventEventDetails`:
 
 ```TS
-private doSomething(CustomEvent e): void {
-    ...
+export interface DelegatedEventEventDetails {
+    originalEvent: Event;
 }
-
 ```
 
-The exact pattern for how we expect event APIs to be implemented is TBD. The above is simply illustrative of one approach, but it's safe to say that the goal will be to provide frameworks like Angular the expected event binding APIs.
--->
+A column type (or an Angular directive on that type) may register a listener for the `delegated-event` event. This listener will have access to the full original event, including the originating cell view via `event.target`. The cell view can expose anything necessary from its public API.
+
+```TS
+class AnchorColumnDirective {
+    constructor(private elementRef: ElementRef) {
+        elementRef.nativeElement.addEventListener('delegated-event', event => {
+            const originalEvent = (event as CustomEvent).details.originalEvent;
+            if (originalEvent.type !== 'click') {
+                return;
+            }
+
+            if ((originalEvent as MouseEvent).button !== 0) {
+                return;
+            }
+
+            const cellView = originalEvent.target as TableColumnAnchorCellView;
+            this.doSomething(cellView.anchor.href);
+            originalEvent.preventDefault();
+        });
+    }
+}
```

Comment:
It's just a normal event, I'd expect a normal `@HostListener` in the example (more interested in knowing if that is not possible for some reason)

```ts
class AnchorColumnDirective {
  @HostListener('delegated-event', ['$event.details.originalEvent'])
  onDelegatedEvent(originalEvent: CustomEvent<DelegatedEventEventDetails>) {
      if (originalEvent.type !== 'click') {
          return;
      }
      if ((originalEvent as MouseEvent).button !== 0) {
          return;
      }
      const cellView = originalEvent.target as TableColumnAnchorCellView;
      this.doSomething(cellView.anchor.href);
      originalEvent.preventDefault();
  }
}
```

Reply:
Nope, that should work. Updated.